homeassistant/components/sensor/teksavvy.py

```diff
@@ -0,0 +1,168 @@
+"""
+Support for TekSavvy.
+
+Get bandwidth comsumption data using Teksavvy API
+You can get your API key here:
+https://myaccount.teksavvy.com/ApiKey/ApiKeyManagement
+
+TekSavvy only counts download only as part of the bandwidth
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.teksavvy/
+"""
+import logging
+from datetime import timedelta
+import asyncio
+import async_timeout
+import voluptuous as vol
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.entity import Entity
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+from homeassistant.const import (
+    CONF_API_KEY, CONF_NAME, CONF_MONITORED_VARIABLES)
+
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'TekSavvy'
+CONF_TOTAL_BANDWIDTH = 'total_bandwidth'
+
+GIGABYTES = 'GB'  # type: str
+PERCENT = '%'  # type: str
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(hours=1)
+REQUEST_TIMEOUT = 5  # seconds
+
+SENSOR_TYPES = {
+    'usage': ['Usage', PERCENT, 'mdi:percent'],
+    'usage_gb': ['Usage', GIGABYTES, 'mdi:download'],
+    'limit': ['Data limit', GIGABYTES, 'mdi:download'],
+    'onpeak_download': ['On Peak Download', GIGABYTES, 'mdi:download'],
+    'onpeak_upload': ['On Peak Upload ', GIGABYTES, 'mdi:upload'],
+    'onpeak_total': ['On Peak Total', GIGABYTES, 'mdi:download'],
+    'offpeak_download': ['Off Peak download', GIGABYTES, 'mdi:download'],
+    'offpeak_upload': ['Off Peak Upload', GIGABYTES, 'mdi:upload'],
+    'offpeak_total': ['Off Peak Total', GIGABYTES, 'mdi:download'],
+    'onpeak_remaining': ['Remaining', GIGABYTES, 'mdi:download']
+}
+
+API_HA_MAP = (
+    ('OnPeakDownload', 'onpeak_download'),
+    ('OnPeakUpload', 'onpeak_upload'),
+    ('OffPeakDownload', 'offpeak_download'),
+    ('OffPeakUpload', 'offpeak_upload'))
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_MONITORED_VARIABLES):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_TOTAL_BANDWIDTH): cv.string,
```

Comment:
Only allow integers here. This way you don't need the casting on line 73.

Reply:
Done. Allows only positive integer now.